src/ReactiveUI/POCOObservableForProperty.cs

```diff
@@ -27,9 +27,9 @@ public int GetAffinityForObject(Type type, string propertyName, bool beforeChang
         {
             var type = sender.GetType();
             if (!hasWarned.ContainsKey(type)) {
-                this.Log().Warn(
-                    "{0} is a POCO type and won't send change notifications, WhenAny will only return a single value!",
-                    type.FullName);
+                var info = expression.GetMemberInfo();
```

Comment:
Is it worth nullchecking expression here? 

Also, there's potentially a little oddity in the way the hasWarned cache is used (you'd get a notification for the first property of the type that's bound to, but no more) I think it's probably fine though :)

Reply:
Yeah, I use a Tuple for storing if we have seen the value before now. 

Also the expression.GetMemberInfo() does have a performance hit and the code that gets the observable for the WhenAny code etc already passed the GetMemberInfo().Name into another function so now I cache it. As you can see in the bench marking below it reduces the time per call but is a wide spread change now.